src/number_theory/pythagorean_triples.lean

```diff
@@ -0,0 +1,546 @@
+/-
+Copyright (c) 2020 Paul van Wamelen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Paul van Wamelen.
+-/
+import algebra.field
+import algebra.gcd_domain
+import algebra.group_with_zero_power
+import tactic
+
+/-!
+# Pythagorean Triples
+The main result is the classification of pythagorean triples. The final result is for general
```

Comment:
I'd love to see more written here about the approach. This is an interesting and understandable file; I can imagine there will be people looking at it who would like some guidance.

Note the information about header comments in the [doc guidelines](https://leanprover-community.github.io/contribute/doc.html) and see some lovely examples about [manifolds](https://github.com/leanprover-community/mathlib/blob/master/src/geometry/manifold/charted_space.lean)! Not that there's necessarily that much to say here, but I'd love to read a natural language description of what you've done.